sklearn/model_selection/_subselect.py

```diff
@@ -0,0 +1,573 @@
+                "cv_results_ must be a dict of fitted GridSearchCV or RandomSearchCV"
+                " objects."
+            )
+
+        _splits = [
+            i
+            for i in list(self.cv_results_.keys())
+            if f"test_{self.scoring}" in i and i.startswith("split")
+        ]
+        if len(_splits) == 0:
+            raise KeyError(f"Scoring metric {self.scoring} not found in cv grid.")
+        else:
+            return _splits
+
+    @property
+    def _n_folds(self):
+        # Extract number of folds from cv_results_. Note that we cannot get this from
+        # the `n_splits_` attribute of the `cv` object because it is not exposed to the
+        # refit callable.
+        return len(
+            list(
+                set(
+                    [
+                        i.split("_")[0]
+                        for i in list(self.cv_results_.keys())
+                        if i.startswith("split")
+                    ]
+                )
+            )
+        )
+
+    @property
+    def _score_grid(self):
+        # Extract subgrid corresponding to the scoring metric of interest
+        return np.vstack([self.cv_results_[cv] for cv in self._get_splits()]).T
+
+    @property
+    def _cv_means(self):
+        # Calculate means of subgrid corresponding to the scoring metric of interest
+        return np.array(np.nanmean(self._score_grid, axis=1))
+
+    @property
+    def _lowest_score_idx(self):
+        # Return index of the lowest performing model
+        return np.nanargmin(self._cv_means)
+
+    @property
+    def _best_score_idx(self):
+        # Return index of the highest performing model
+        return np.nanargmax(self._cv_means)
+
+    def _apply_thresh(self, param: str, l_cutoff: float, h_cutoff: float) -> int:
+        if not any(param in x for x in self.cv_results_["params"][0].keys()):
+            raise KeyError(f"Parameter {param} not found in cv grid.")
+        else:
+            hyperparam = [
+                i for i in self.cv_results_["params"][0].keys() if i.endswith(param)
+            ][0]
+
+        # Mask out all grid columns that are outside the performance window
+        self.cv_results_[f"param_{hyperparam}"].mask = np.where(
+            (self._cv_means >= float(l_cutoff)) & (self._cv_means <= float(h_cutoff)),
+            True,
+            False,
+        )
+
+        if np.sum(self.cv_results_[f"param_{hyperparam}"].mask) == 0:
+            print(
+                f"\nLow: {l_cutoff}\nHigh: {h_cutoff}\nMeans across folds:"
+                f" {self._cv_means}\nhyperparam: {hyperparam}\n"
+            )
+            raise ValueError(
+                "No valid grid columns remain within the boundaries of the specified"
+                " performance window."
+            )
+
+        # Find the highest surviving rank (i.e. the lowest performing model among those
+        # remaining within the performance window)
+        highest_surviving_rank = np.nanmax(
```

Comment:
Can `nanargmax` be used? to avoid the {flatnonzero, isin} below?

Reply:
It sure can :-)